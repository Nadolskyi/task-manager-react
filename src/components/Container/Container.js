import React, { Component } from 'react'
import { connect } from 'react-redux'
import './Container.css'
import ListTask from '../ListTask/ListTask'
import AddTask from '../AddTask/AddTask'
import { getTasks } from '../../actions/actions'

export class Container extends Component {

  componentDidMount() {
    this.props.onGetTasks();
  }

  render() {
    return (
      this.props.tasks.length ? <div className="container Container-marginTop Container-color">
        <AddTask />
        <ListTask tasks = {this.props.tasks} />
      </div> : <h1>Loading...</h1>
    );
  }
}

const mapStateToProps = state => ({
  tasks: state
});

const mapDispatchToProps = dispatch => ({
  onGetTasks: () => setTimeout(() => dispatch(getTasks()), 1000),
})

export default connect(mapStateToProps, mapDispatchToProps)(Container)