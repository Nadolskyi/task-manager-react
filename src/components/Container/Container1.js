import React from 'react'
import { shallow } from 'enzyme'
import { Container } from './Container'

describe('Container', () => {
  const props = { // описываем props
    tasks: [],
    // функция получения новостей onGetNews. Ее содержимое тестировать не нужно.
    // Но нам потребуется протестировать, что функция была вызвана в componentDidMount
    onGetTasks: () => {},
  }
  
  // здесь будут будущие it
  
})