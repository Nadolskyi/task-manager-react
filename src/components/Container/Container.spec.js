import React from 'react'
import { shallow } from 'enzyme'
import { Container } from './Container'
import '../../setupTest'

describe('Container', () => {
  const props = {
    tasks: [],
    onGetTasks: () => { },
  }

  describe('container loading', () => {
    const container = shallow(<Container {...props} />)

    it('renders properly', () => {
      expect(container).toMatchSnapshot();
    })
    
    it('isLoading', () => {
      expect(container.find('h1').text()).toEqual('Loading...')
    })
  })

  describe('news container render <News List />', () => {
    const nextProps = {
      ...props,
      tasks: [1, 2]
    }
    const container = shallow(<Container {...nextProps} />)

    it('renders properly News List ', () => {
      expect(container).toMatchSnapshot();
    })
    it('render <News List />', () => {
      expect(container.find('ListTask')).toHaveLength(1)
    })
  })

  describe('function has been called', () => {
    const getTask = jest.fn();
    const nextProps = {
      ...props,
      onGetTasks: getTask
    }
    const container = shallow(<Container {...nextProps} />)

    it('been called', () => {
      expect(getTask).toHaveBeenCalled();
    })
  })

})