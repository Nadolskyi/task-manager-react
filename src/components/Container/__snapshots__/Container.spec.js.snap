// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container container loading renders properly 1`] = `ShallowWrapper {}`;

exports[`Container news container render <News List /> renders properly News List  1`] = `ShallowWrapper {}`;
